Automattic/jetpack
# Simple payments: Hide paypal when invalid

When the paypal block was misconfigured it was still rendered on the
front-end. When people tried to use it they would receive an opaque
error message from paypal.

To resolve this, the paypal block will not be rendered on the front end
if it's misconfigured. There are plenty of messages displayed in the
editor if it is misconfigured so post-editors should be able to see any
problems before they happen. This makes it more consistent with the
donations block which does not display anything if not connected to
stripe.

Fixes #17641
#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:






1. Start up jetpack, jurassic tube etc or a wpcom site
2. Apply these changes
3. Ensure your site has either a wpcom premium or above plan or a jetpack complete plan
4. Create a new post
5. Add a Pay with paypal block
6. Without filling the details in, press Update
7. View the post in the front-end
8. You should see that the block does not appear
9. Repeat omitting various details e.g. title, email address, or invalid details e.g. negative currency, invalid email
10. Edit the post again, get the productId from code mode <img width="965" alt="Markup 2022-01-13 at 16 38 40" src="https://user-images.githubusercontent.com/93301/149371277-4f8155d2-5ef5-4e53-9b43-cc64dda898bc.png">
11. Add a shortcode block with the contents like `[simple-payment id="THE_PRODUCT_ID"]Testing123[/simple-payment]`
12. View the post again, ensure the shortcode version renders too.
